util/svdgen.py

```diff
@@ -0,0 +1,604 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Generate SVD file from validated register JSON tree
```

Comment:
This comment is probably where you should have the link to the CMSIS-SVD docs.  I'd also try to minimize the API here, right now there are things that I don't think make sense to call in isolation.